packages/router/utils.ts

```diff
@@ -201,7 +201,7 @@ type _PathParam<Path extends string> =
     : // find params after `:`
     Path extends `:${infer Param}`
     ? Param extends `${infer Optional}?`
-      ? Optional
+      ? `${Optional}?`
```

Comment:
This isn't a separate type, as you've defined below. It's declared on the previous line with `infer`.